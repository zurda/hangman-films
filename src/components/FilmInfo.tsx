import React from 'react'
import styled from '@emotion/styled'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: 'Roboto';
  @media (min-width: 420px) {
    flex-direction: row;
    justify-content: space-between;
  }
`

const Info = styled.p`
  margin: 16px 32px;
`

const Img = styled.img`
  width: 154px;
`

const FilmInfo = ({ film }: any) => (
  <Container>
    <Img
      src={`http://image.tmdb.org/t/p/w154${film.poster_path}`}
      alt="Film poster"
    />
    {film.overview || film.tagline ? (
      <Info>{film.overview || film.tagline}</Info>
    ) : null}
  </Container>
)

export default FilmInfo
