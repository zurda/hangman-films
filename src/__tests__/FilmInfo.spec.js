import React from 'react'
import { create } from 'react-test-renderer'
import FilmInfo from '../components/FilmInfo'

const filmData = {
  overview: 'Some text',
  poster_path: 'sometext.png',
  tagline: 'Some more text'
}

describe('FilmInfo component', () => {
  test('Matches the snapshot', () => {
    const filmInfo = create(<FilmInfo film={filmData} />)
    expect(filmInfo.toJSON()).toMatchSnapshot()
  })
})
