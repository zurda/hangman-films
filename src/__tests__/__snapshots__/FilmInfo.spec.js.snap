// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfo component Matches the snapshot 1`] = `
<div
  className="css-9s8wn8"
>
  <img
    alt="Film poster"
    className="css-rh3cin"
    src="http://image.tmdb.org/t/p/w154sometext.png"
  />
  <p
    className="css-bni6fn"
  >
    Some text
  </p>
</div>
`;
